R/addAlpha.R

```diff
@@ -585,6 +585,19 @@ setMethod("addAlpha", signature = c(x = "SummarizedExperiment"),
         FUN("'faith' index can be calculated only for TreeSE with rowTree(x) ",
             "populated.", call. = FALSE)
     }
+    # Check for unsupported values (relative or negative)
+    if(any(assay(x) < 0)){
+        unsupported_indices <- detected[["index"]]
+        detected <- detected[!detected[["index"]] %in% unsupported_indices, ]
+        if(length(unsupported_indices) > 0){
+            warning_indices <- paste0(unsupported_indices, collapse = "', '")
+            warning_indices <- paste0("'", warning_indices, "'")
+            warning(
+                "The following indices cannot be calculated due to unsupported 
+                values (negative values): ",
+                warning_indices, call. = FALSE)
+        }
+    }
```

Comment:
1.

What if there is no indices to calculate anymore? All were removed due to negative values? This will fail then. Check line 584 above.

2.

If there is negative values, you are subsetting the index data.frame by taking those indices that are not in that data.frame. That does not make sense.

-->
- Is there any index that can handle negative values?
```

if( any(assay(x, assay.type) < 0) ){
    indices_that_can_handle_neg_vals <- c("test")
    ind <- detected[["index"]] %in% indices_that_can_handle_neg_vals
    index_rm <- detected[ind, "index"]
    detected <- detected[ind, ]
    if( length(index_rm) > 0 ){
        FUN <- if( nrow(detected) == 0 ) stop else warning
        FUN(
                "The following indices cannot be calculated due to unsupported 
                values (negative values): ",
                warning_indices, call. = FALSE)
    }
}

```

Reply:
Or ore preferably, you could add negative value info to index data.frame in the beginning of this function

# Supported diversity indices
    temp <- c(
        "coverage", "faith", "fisher", "gini_simpson", "inverse_simpson",
        "log_modulo_skewness", "shannon")
    temp <- data.frame(index = temp)
    temp[["measure"]] <- "diversity"
    temp[["index_long"]] <- paste0(temp[["index"]], "_", temp[["measure"]])
    temp[["FUN"]] <- ".estimate_diversity"
   # Can handle negative values
   temp[["non_neg"]] <- c(TRUE, TRUE, ... FALSE)
    supported[["diversity"]] <- temp